### PR TITLE
feat: Fetch items-list for user from server

### PR DESCRIPTION
**Pull request by ashrafmahmood**
#### Short description of what this resolves:
Fetch items-list for user from server and add loading and error UI.

#### Changes proposed in this pull request:
- Fixes deepsource error
- Replace localStorage with window.localStorage

#### After

https://user-images.githubusercontent.com/55829865/121376101-ddc95480-c95e-11eb-9a30-b8585c489c3f.mp4

#### Checklist

- [x] My branch is up-to-date with the Upstream `main` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ashrafmahmood** [comment]
@Haider8  Please check this PR

**Comment by Haider8** [comment]
@ashrafmahmood Add tests also

**Comment by ashrafmahmood** [comment]
https://user-images.githubusercontent.com/55829865/121867439-973c7700-cd1d-11eb-8e2f-0c880b462377.mp4

File: src/components/Home/Shop.js [hunk 1]
```diff
@@ -31,12 +31,14 @@ function Shop ({ address, loading, name }) {
 
 Shop.propTypes = {
   address: PropTypes.string,
+  id: PropTypes.number,
```
Comment (Haider8) [inline src/components/Home/Shop.js:34]: Make this a string now as I am making a slug for id using shop name

File: src/components/Home/Shop.js [hunk 2]
```diff
@@ -31,12 +31,14 @@ function Shop ({ address, loading, name }) {
...
   loading: PropTypes.bool.isRequired,
   name: PropTypes.string
 }
 
 Shop.defaultProps = {
   address: '',
+  id: null,
```
Comment (Haider8) [inline src/components/Home/Shop.js:41]: Please explain why the default is null?

**Review by Haider8 — CHANGES_REQUESTED** [review]

**Comment by Haider8** [comment]
@ashrafmahmood Fix the deepsource issues here and then the PR will get merged

File: src/components/Home/Shop.js (hunk 2 above)
Comment (ashrafmahmood) [reply src/components/Home/Shop.js:41]: Should I make it a empty string?

**Pull request merged**